### ISSUE TITLE
Can the if() condition be asynchronous?

### ISSUE THREAD

**Issue by jbcpollak** [Reactive-Extensions/RxJS#1095]
It doesn't appear to be the case according to the documentation, is there any reason the condition for the `if()` method can't be asynchronous?

    var source = Rx.Observable.if(
        Rx.Observable.return(shouldRun),
        Rx.Observable.return(42)
    );

or if necessary:

    var source = Rx.Observable.if(
        function() { return Rx.Observable.return(shouldRun)},
        Rx.Observable.return(42)
    );

**Comment by mattpodwysocki** [comment]
@jbcpollak no, there is no reason for why it is synchronous.  What is the use case for it being async?

**Comment by jbcpollak** [comment]
I made this implementation:

```
function asyncIf(conditionalSource, thenSource, elseSource) {

	return conditionalSource
		.flatMap(conditional =>
			Rx.Observable.if(
				() => conditional,
				thenSource,
				elseSource
			)
		);
}
```

Based on a stream of inputs, I select between two other streams
